recipe_scrapers/_schemaorg.py

```diff
@@ -25,50 +65,40 @@ def __init__(self, page_data, raw=False):
-            syntax_data = data.get(syntax, [])
-            try:
-                index = [x.get("@type", "") for x in syntax_data].index("Recipe")
-                syntax_data.insert(0, syntax_data.pop(index))
-            except ValueError:
-                pass
-
-            for item in syntax_data:
-                in_context = SCHEMA_ORG_HOST in item.get("@context", "")
-                item_type = item.get("@type", "")
-                if isinstance(item_type, list):
-                    for type in item_type:
-                        if type.lower() in low_schema:
-                            item_type = type.lower()
-                if in_context and item_type.lower() in low_schema:
-                    self.format = syntax
-                    self.data = item
-                    if item_type.lower() == "webpage":
-                        self.data = self.data.get("mainEntity")
+        extractor = Extractor(page_data)
+        for element in extractor.linked_data:
+            if isinstance(element, dict) and "@graph" in element:
+                graph_nodes = element["@graph"]
+            elif isinstance(element, list):
+                graph_nodes = element
+            else:
+                graph_nodes = [element]
+            for node in graph_nodes:
+                node_type = str().join(node.get("@type", [])).lower()
+                if node_type == "recipe":
+                    self.data = node
                     return
-                elif in_context and "@graph" in item:
-                    for graph_item in item.get("@graph", ""):
-                        graph_item_type = graph_item.get("@type", "")
-                        if not isinstance(graph_item_type, str):
-                            continue
-                        if graph_item_type.lower() in low_schema:
-                            in_graph = SCHEMA_ORG_HOST in graph_item.get("@context", "")
-                            self.format = syntax
-                            if graph_item_type.lower() == "webpage" and in_graph:
-                                self.data = self.data.get("mainEntity")
-                                return
-                            elif graph_item_type.lower() == "recipe":
-                                self.data = graph_item
-                                return
+
+        microdata = defaultdict(list)
+        for element in extractor.microdata:
+            if "recipe" in element.attrib["itemtype"].lower():
+                for node in element.findall(
+                    path=".//html:*[@itemprop]",
+                    namespaces={"html": "http://www.w3.org/1999/xhtml"},
+                ):
+                    name = node.attrib["itemprop"]
+                    if "content" in node.attrib:
+                        value = node.attrib["content"]
+                    else:
+                        value = str().join(chunked_text(node)).strip()
+                    microdata[name].append(value)
```

Comment:
This is a completely naive and by no-means spec-adhering way to collect microdata from the page.

One key situation that it doesn't handle is nested microdata objects (where `itemscope` appears as an HTML attribute at various levels in a tree of HTML elements).